sopel/db.py

```diff
@@ -189,6 +194,7 @@ def get_nick_id(self, nick, create=True):
                 nickname = Nicknames(nick_id=nick_id.nick_id, slug=slug, canonical=nick)
                 session.add(nickname)
                 session.commit()
+            self.nick_id_lock.release()
```

Comment:
If an uncaught exception is raised anywhere in this relatively long `try` block, the lock might never be released. That would be Very Bad™. Releasing the lock should be done in the `finally` clause, below.

Reply:
moved to `finally`